wasm/store.go

```diff
@@ -844,11 +858,28 @@ func DecodeBlockType(types []*TypeInstance, r io.Reader) (*FunctionType, uint64,
 
 // HostFunctionCallContext is the first argument of all host functions.
 type HostFunctionCallContext struct {
+	ctx context.Context
 	// Memory is the currently used memory instance at the time when the host function call is made.
 	Memory *MemoryInstance
 	// TODO: Add others if necessary.
 }
 
+// NewHostFunctionCallContext creates a new HostFunctionCallContext with a
+// context and memory instance.
+func NewHostFunctionCallContext(ctx context.Context, memory *MemoryInstance) *HostFunctionCallContext {
+	return &HostFunctionCallContext{
+		ctx:    ctx,
+		Memory: memory,
+	}
+}
+
+// Context returns the host call's context. To change the context, use WithContext.
```

Comment:
Maybe leave this out for now, unless you want to add and test WithContext now

```suggestion
// Context returns the host call's context.
```

Reply:
Nope. Probably not worth it or add any value :)